src/bluesearch/database/topic_info.py

```diff
@@ -0,0 +1,116 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+"""Implementation of the TopicInfo data structure."""
+from __future__ import annotations
+
+import copy
+import datetime
+import pathlib
+from dataclasses import dataclass, field
+from typing import Any
+
+import bluesearch
+from bluesearch.database.article import ArticleSource
+
+
+@dataclass
+class TopicInfo:
+    """The topic information extracted from a journal article.
+
+    For the spec see the following GitHub issue/comment:
+    https://github.com/BlueBrain/Search/issues/518#issuecomment-985525160
+    """
+
+    source: ArticleSource
+    path: str | pathlib.Path
+    element_in_file: int | None = None
+    article_topics: dict[str, list[str]] = field(init=False, default_factory=dict)
+    journal_topics: dict[str, list[str]] = field(init=False, default_factory=dict)
+    metadata: dict = field(default_factory=dict)
```

Comment:
I don't like this metadata either. I was following the spec here https://github.com/BlueBrain/Search/issues/518#issuecomment-985525160 where the idea of metadata was to contain any source-specific information.

I can actually remove it now because we don't have anything source-specific at the moment. Note though that the `metadata` will still be part of the dict returned by `TopicInfo.json()` and will contain a dictionay with `created-date`, `bbs-version`, and an optional `element_in_file` since this is what we put in there [on master](https://github.com/BlueBrain/Search/blob/d41611ad516e05b2fe28007659b4f33f3d6ce1a1/src/bluesearch/entrypoint/database/topic_extract.py) by hand.

Reply:
Removed the `metadata` parameter in 827c79d